### PR TITLE
Seqr - memory optimisation - only add missing records to the list of entries to be updated.

### PR DESCRIPTION
**Pull request by milo-hyben**
Background, when trying to process batch https://batch.hail.populationgenomics.org.au/batches/430882
, which took over a week to process, the function runs out of memory because it is trying to prepare the list of all gcp records for that time. Even most of them would be already upserted as seqr scheduled aggregation runs every 4 hours.

This PR checks if records exist and only add it to a list if not present.
This should minimise the list passed around and therefore minimising function RAM requirements.

### CODE REVIEW

File: cpg_infra/billing_aggregator/aggregate/seqr.py [hunk 1]
```diff
@@ -395,8 +395,16 @@ def generator() -> Generator[dict, None, None]:
 
             json_objs_iter = generator()
 
+    def _append_if_not_present(entries: list[dict], obj: dict) -> None:
+        """
+        Append records if not already in the table
+        """
+        if obj['id'] not in existing_ids:
+            # only insert if it's not already in the table
+            entries.append(obj)
```
Comment (illusional) [inline cpg_infra/billing_aggregator/aggregate/seqr.py:404]: Maybe a nitpick, but I don't think you need this function, it's logically pretty simple (not complex enough for an abstraction), what do you think?

**Review by illusional — APPROVED** [review]

File: cpg_infra/billing_aggregator/aggregate/seqr.py (hunk 1 above)
Comment (milo-hyben) [reply cpg_infra/billing_aggregator/aggregate/seqr.py]: Good point, saving 2 lines in the loop can be done differently.

File: cpg_infra/billing_aggregator/aggregate/seqr.py [hunk 2]
```diff
@@ -443,14 +443,15 @@ def generator() -> Generator[dict, None, None]:
...
 
             # For every seqr billing entry migrate it over
-            entries.append(obj)
-            entries.append(
-                utils.get_credit(
-                    entry=obj,
-                    topic='seqr',
-                    project=utils.SEQR_PROJECT_FIELD,
-                ),
+            entries.append(obj) if obj['id'] not in existing_ids else None
```
Comment (illusional) [inline cpg_infra/billing_aggregator/aggregate/seqr.py:446]:
```suggestion
            if obj['id'] not in existing_ids:
                entries.append(obj)
```

Comment (milo-hyben) [reply cpg_infra/billing_aggregator/aggregate/seqr.py]: Yes, that one was my original, than I did not like the main loop does not fit on the screen.
But I guess it reads better.

**Review by illusional — APPROVED** [review]

**Pull request merged**